composition_batch.go

```diff
@@ -17,7 +17,11 @@ func (c *BatchComposition) Execute(m *Message) (resp *Response, err error) {
 
 	do := func(do recfunc, m *Message) {
 		cmd := c.lpop()
-		resp, err = performJob(cmd, m)
+
+		err = try(c.engine.retryConf, func() (err error) {
+			resp, err = performJob(cmd, m)
```

Comment:
Why not alter performJob itself and have it handle the retries internally?
I mean change the parameters to whatever you want to pass. Will be better than this copy-pasting no?